debezium-embedded/src/main/java/io/debezium/embedded/AsyncEngineConfig.java

```diff
@@ -0,0 +1,85 @@
+                    + "'UNORDERED' means non-sequential processing, i.e. the records can be produced in a different order than the original one. "
+                    + "Non-sequential approach gives better throughput, as the records are produced immediately once the SMTs and serialization of "
+                    + "the message is done, without waiting of other records. This option doesn't have any effect when ChangeConsumer is provided to the engine.")
+            .withDefault("ORDERED");
+
+    /**
+     * An optional field that specifies if the default {@link io.debezium.engine.DebeziumEngine.ChangeConsumer} should be created for consuming records or not.
+     * If only {@link java.util.function.Consumer} is provided to the engine and this option is set to {@code true} (the default is {@code false}), engine will create default
+     * {@link io.debezium.engine.DebeziumEngine.ChangeConsumer} and use it for record processing. Default {@link io.debezium.engine.DebeziumEngine.ChangeConsumer}
+     * implementation is taken from legacy {@link EmbeddedEngine}, so this option allows to use almost the same implementation for record processing as {@link EmbeddedEngine}.
+     * The only difference to {@link EmbeddedEngine} is that SMTs will be still run in parallel, even when this option is turned on.
+     * This option doesn't have any effect when {@link io.debezium.engine.DebeziumEngine.ChangeConsumer} is already provided to the engine in the configuration.
+     */
+    Field RECORD_PROCESSING_WITH_SERIAL_CONSUMER = Field.create("record.processing.with.serial.consumer")
+            .withDescription("Specifies whether the default ChangeConsumer should be created from provided Consumer, resulting in serial Consumer processing. "
+                    + "This option has no effect if the ChangeConsumer is already provided to the engine via configuration.")
+            .withDefault(false)
+            .withValidation(Field::isBoolean);
+
+    /**
+     * An internal field that specifies the maximum amount of time to wait for a task lifecycle operation, i.e. for starting and stopping the task.
+     */
+    Field TASK_MANAGEMENT_TIMEOUT_MS = Field.createInternal("task.management.timeout.ms")
+            .withDescription("Time to wait for task's lifecycle management operations (starting and stopping), given in milliseconds. "
+                    + "Defaults to 2 minutes (120_000 ms).")
+            .withDefault(120_000L)
+            .withValidation(Field::isPositiveInteger);
+
+    /**
+     * The array of all exposed fields.
+     */
+    Field.Set ALL_FIELDS = EmbeddedEngineConfig.ALL_FIELDS.with(
+            RECORD_PROCESSING_SHUTDOWN_TIMEOUT_MS,
+            RECORD_PROCESSING_THREADS,
+            RECORD_PROCESSING_ORDER,
+            RECORD_PROCESSING_WITH_SERIAL_CONSUMER,
+            // internal fields
+            TASK_MANAGEMENT_TIMEOUT_MS);
```

Comment:
It is not always suggested to add internal fields to `ALL_FIELDS`, but sometimes we do. Is this intentional here?

Reply:
TBH I'm not aware about any policy or system what should or shouldn't be added in `ALL_FIELDS`. So this is the result of my lack of knowledge in this area and therefore I added everything. So I get it right `TASK_MANAGEMENT_TIMEOUT_MS` should be removed from there?

Reply:
Hi @rk3rn3r, I have not looked at this PR in it's totality to know precisely how this `ALL_FIELDS` are used, so I'm going to rely on @vjuranek to help me out here :)

At least in KC land where we define a connector, we include the internal properties as part of the `ALL_FIELDS` declaration as this is used to define what the `ConfigDef` is that gets returned to KC and ultimately what KC bases it's own validation hooks upon.  So if we omit such fields from the `ALL_FIELDS` configuration, that field won't have it's validation hooks fired at least within KC -- and I'd surmise that we ideally should be operating in very much the same approach in the Embedded Engine and by Extension DS.

Reply:
In fact we do add them to `ALL_FIELDS` in connectors - at least I found most of them. It is possible that we are not really consistent. But we add `InvisibleRecommender` to them. This should be in fact done by default in `createInternal` instead of delegating it to field definitions.